### PR TITLE
RB: add model for the `Digest` and `OpenSSL::Digest` modules

### PR DESCRIPTION
**Pull request by erik-krogh**
CVE-2019-5420: TP/TN

[An evaluation](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/PR-10832-0-ruby/reports) shows good performance.  
The `rb/weak-cryptographic-algorithm` query has many new results with this change, but the results seem reasonable given the query.

### CODE REVIEW

File: ruby/ql/lib/codeql/ruby/frameworks/core/Digest.qll [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/** A call that hashes some input using a hashing algorithm from the `Digest` module. */
+private class DigestCall extends Cryptography::CryptographicOperation::Range instanceof DataFlow::CallNode {
+  Cryptography::HashingAlgorithm algo;
+
+  DigestCall() {
+    this = digest(algo).getAMethodCall(["hexdigest", "base64digest", "bubblebabble"])
+    or
+    this = digest(algo).getAMethodCall("file") // it's directly hashing the contents of a file, but that's close enough for us.
+    or
+    this = digest(algo).getMethod("new").getReturn().getAMethodCall(["digest", "update", "<<"])
```
Comment (aibaars) [inline ruby/ql/lib/codeql/ruby/frameworks/core/Digest.qll:26]: Isn't `getMethod("new").getReturn()` equivalent to `getAnInstance()` ?

**Review by aibaars — APPROVED** [review]
Looks good to me.

File: ruby/ql/lib/codeql/ruby/frameworks/core/Digest.qll (hunk 1 above)
Comment (erik-krogh) [reply ruby/ql/lib/codeql/ruby/frameworks/core/Digest.qll]: It is 👍

**Review by aibaars — APPROVED** [review]
Looks good to me.

**Pull request merged**